src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -738,7 +738,8 @@ CHIP_ERROR DefaultSceneTableImpl::SceneSaveEFS(SceneTableEntry & scene)
     if (!HandlerListEmpty())
     {
         uint8_t clusterCount = 0;
-        clusterId cArray[kMaxClustersPerScene];
+        // TODO: Use dynamical allocation with emberAFClusterCount to generate a smaller array
+        clusterId cArray[GENERATED_CLUSTER_COUNT];
```

Comment:
Why these changes?  These have nothing to do with on/off.  They should be in a separate PR with a separate clear description of what they are aiming at.

Reply:
Because the number of clusters in an endpoint is greater than the maximum clusters allowed per scenes, which means if we don't iterate over all the clusters of an endpoint, it is possible to not hit it's sceneable ones (which is what happened during tests)

This will be moved to a separate PR.